ddtrace/internal/sampling.py

```diff
@@ -82,3 +95,74 @@ def update_sampling_decision(
         return _set_trace_tag(context, sampling_mechanism)
     else:
         return _unset_trace_tag(context)
+
+
+class SpanSamplingRule:
+    """A span sampling rule to evaluate and potentially tag each span upon finish."""
+
+    __slots__ = (
+        "service_matcher",
+        "name_matcher",
+        "sample_rate",
+        "max_per_second",
+        "sampling_id_threshold",
+        "limiter",
+        "matcher",
+    )
+
+    def __init__(
+        self,
+        service=None,  # type: Optional[str]
+        name=None,  # type: Optional[str]
+        sample_rate=1.0,  # type: Optional[float]
+        max_per_second=None,  # type: Optional[int]
+    ):
+        self.set_sample_rate(sample_rate)
+        self.max_per_second = max_per_second
```

Comment:
Do each of these attributes have to be public?

Reply:
No, will change.